### PR TITLE
Sql: implement RENAME COLUMN

### LINKED ISSUE #2034: sql: add support for ALTER [DATABASE|INDEX|TABLE] [IF EXISTS] name RENAME [COLUMN] TO
The support exists within the grammar, though we need to flesh out the syntax tree elements for these statements. The existing `RenameTable` struct was inherited from the vitess code and will need to change (it isn't currently referenced by the grammar). Actual implementation of these statements should be straightforward.

### PR DESCRIPTION
**Pull request by wenyong-h**
Close #2034.

### CODE REVIEW

File: sql/rename.go [hunk 1]
```diff
@@ -209,3 +209,78 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
...
+
+	for _,idx := range tableDesc.Indexes {
+		for i, id := range idx.ColumnIDs {
+			if id == column.ID {
+				idx.ColumnNames[i] = newColName
+			}
+		}
+	}
+	column.Name = newColName
+
```
Comment (MBerhault) [inline sql/rename.go]: add a call to tableDesc.Validate just before writing it.

File: sql/rename.go [hunk 2]
```diff
@@ -209,3 +209,78 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
 
 	return &valuesNode{}, nil
 }
+
+// RenameColumn renames the column.
+// Privileges: CREATE on table.
```
Comment (petermattis) [inline sql/rename.go]: @MBerhault Is this the correct privilege?

File: sql/rename.go [hunk 3]
```diff
@@ -209,3 +209,78 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
...
+		return nil, err
+	}
+
+	dbDesc, err := p.getDatabaseDesc(n.Table.Database())
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if table exists.
+	tbKey := tableKey{dbDesc.ID, string(n.Table.Table())}.Key()
```
Comment (petermattis) [inline sql/rename.go]: `n.Table.Table()` returns a `string`. No need to cast to a `string`.

File: sql/rename.go (hunk 2 above)
Comment (MBerhault) [inline sql/rename.go]: yup. Anything modifying the schema requires the same privileges as creating it (although mysql also has a `ALTER` privilege, but we don't).

File: sql/rename.go [hunk 4]
```diff
@@ -209,3 +209,78 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
...
+
+	tableDesc, err := p.getTableDesc(n.Table)
+	if err != nil {
+		return nil, err
+	}
+
+	colName := string(n.Name)
+	column, err := tableDesc.FindColumnByName(colName)
+	if err != nil {
+		return nil, err
```
Comment (petermattis) [inline sql/rename.go]: Should the `n.IfExists` apply here as well?

**Comment by petermattis** [comment]
LGTM modulo a few nits.

File: sql/rename.go (hunk 1 above)
Comment (MBerhault) [inline sql/rename.go]: could you also add it to RenameIndex?

File: sql/rename.go (hunk 4 above)
Comment (wenyong-h) [inline sql/rename.go]: tested on postgres, `if exists` only applies to table.

Comment (petermattis) [inline sql/rename.go]: Ok. Perhaps add a comment here so that we don't try to "fix" this again later.

**Comment by wenyong-h** [comment]
updated.

**Comment by MBerhault** [comment]
LGTM

File: sql/parser/rename.go [hunk 5]
```diff
@@ -71,3 +71,21 @@ func (node *RenameIndex) String() string {
 	_, _ = buf.WriteString(fmt.Sprintf("%s RENAME TO %s", node.Name, node.NewName))
 	return buf.String()
 }
+
+// RenameColumn represents a RENAME COLUMN statement.
+type RenameColumn struct {
+	Table    *QualifiedName
+	Name     Name
+	NewName  Name
+	IfExists bool
```
Comment (tamird) [inline sql/parser/rename.go]: can you add a comment here that this refers to the table?

Comment (wenyong-h) [inline sql/parser/rename.go]: @tamird Added.

File: sql/rename.go [hunk 6]
```diff
@@ -203,6 +203,88 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
 
 	idx.Name = newIdxName
 	descKey := MakeDescMetadataKey(tableDesc.GetID())
+	if err := tableDesc.Validate(); err != nil {
```
Comment (tamird) [inline sql/rename.go]: can you add a test for this, since you added it?

**Comment by tamird** [comment]
LGTM let me know when you've added the test and i'll merge

File: sql/rename.go (hunk 6 above)
Comment (wenyong-h) [inline sql/rename.go]: @tamird how do i test that? I runned a `select` after rename the column, which test that indirectly.
Suggestions?

Comment (MBerhault) [inline sql/rename.go]: I don't think there's anything you can pass that will reach Validate() and not be caught by your code first.

Comment (tamird) [inline sql/rename.go]: oh, my mistake - i was thinking the addition here was a permission check. sorry!

**Pull request merged**